src/env/default.rs

```diff
@@ -257,13 +277,76 @@ impl WriteExt for LogFile {
     }
 }
 
+pub struct AioContext {
+    aio_vec: Vec<aiocb>,
+    pub(crate) buf_vec: Vec<Arc<Mutex<Vec<u8>>>>,
+}
+impl AioContext {
+    pub fn new(block_sum: usize) -> Self {
+        let mut vec = vec![];
+        unsafe {
+            for i in 0..block_sum {
+                vec.push(mem::zeroed::<libc::aiocb>());
+            }
+        }
+        Self {
+            aio_vec: vec,
+            buf_vec: vec![],
+        }
+    }
+}
+
+impl AsyncContext for AioContext {
+    fn single_wait(&mut self, seq: usize) -> IoResult<usize> {
+        let buf_len = self.buf_vec[seq].lock().unwrap().len();
+        unsafe {
+            loop {
+                libc::aio_suspend(
+                    vec![&mut self.aio_vec[seq]].as_ptr() as *const *const aiocb,
+                    1 as i32,
+                    ptr::null::<libc::timespec>(),
+                );
+                if buf_len == aio_return(&mut self.aio_vec[seq]) as usize {
+                    return Ok(buf_len);
+                }
+            }
+        }
+    }
+
+    fn wait(&mut self) -> IoResult<usize> {
+        let mut total = 0;
+        for seq in 0..self.aio_vec.len() {
+            match self.single_wait(seq) {
+                Ok(len) => total += 1,
+                Err(e) => return Err(e),
+            }
+        }
+        Ok(total as usize)
+    }
+
+    fn data(&self, seq: usize) -> Arc<Mutex<Vec<u8>>> {
+        self.buf_vec[seq].clone()
+    }
+}
+
 pub struct DefaultFileSystem;
 
 impl FileSystem for DefaultFileSystem {
     type Handle = LogFd;
     type Reader = LogFile;
     type Writer = LogFile;
 
+    fn read_aio(
+        &self,
+        handle: Arc<Self::Handle>,
+        seq: usize,
+        ctx: &mut AioContext,
```

Comment:
`AioContext` should be a generic type, different file system implementation can have different context.

The code will look like:
```rust
trait WaitData {
  fn wait(&mut self, index: usize) -> &[u8];
}

trait FileSystem {
  type AsyncContext: WaitData;
  fn new_async_context(&self) -> Self::AsyncContext;
}

impl FileSystem for DefaultFileSystem {
  type AsyncContext = AioContext;
  fn new_async_context(&self) {
    AioContext::new()
  }
}
```